src/mallocTracer.cpp

```diff
@@ -29,17 +29,33 @@ static realloc_t _orig_realloc = NULL;
 typedef void (*free_t)(void*);
 static free_t _orig_free = NULL;
 
+typedef int (*posix_memalign_t)(void**, size_t, size_t);
+static posix_memalign_t _orig_posix_memalign = NULL;
+
+typedef void* (*aligned_alloc_t)(size_t, size_t);
+static aligned_alloc_t _orig_aligned_alloc = NULL;
+
+// In musl, posix_memalign calls aligned_alloc and it in turns calls malloc.
+// If in_musl_posix_memalign, do not recordMalloc in aligned_alloc_hook.
+// If in_musl_aligned_alloc, do not recordMalloc in malloc_hook.
+// Do not access thread_local variables in non-musl.
+static bool is_musl = false;
+static thread_local bool in_musl_posix_memalign = false;
+static thread_local bool in_musl_aligned_alloc = false;
```

Comment:
`thread_local` may allocate (using malloc!), so it's problematic to use `thread_local` in `malloc` hooks.
Why not just skip these hooks entirely on musl? Will `malloc_hook` catch these functions?

Reply:
In musl, the next-gen [aligned_alloc implementation](https://git.musl-libc.org/cgit/musl/tree/src/malloc/mallocng/aligned_alloc.c?h=v1.2.2) no longer relies on `malloc`, unlike the current-gen version, which uses `malloc` [in certain cases](https://git.musl-libc.org/cgit/musl/tree/src/malloc/oldmalloc/aligned_alloc.c#n22).

[As of 2012](https://git.musl-libc.org/cgit/musl/commit/src/malloc/posix_memalign.c?id=9bff7c133e73ecfb200614d7a7d386a164a1a61f), `posix_memalign` has consistently called `aligned_alloc` instead of using `malloc`, so hooking `posix_memalign` in musl is unnecessary.

However, since `aligned_alloc` no longer necessarily calls `malloc`, it still requires hooking to ensure proper behavior.

Reply:
Wait, new `aligned_alloc` [does call](https://git.musl-libc.org/cgit/musl/tree/src/malloc/mallocng/aligned_alloc.c#n24) `malloc`, am I wrong?

Reply:
Sorry, yes. I haven't seen this in jfr profiles of `aligned_alloc` calls. Need to check this again.